app/assets/javascripts/form/dates.js.coffee

```diff
@@ -1,15 +1,31 @@
 (($) ->
 
-  unless Modernizr.touch and Modernizr.inputtypes.date
+  unless Modernizr.touch
 
     # Initializes date fields
+    $(document).ready ->
+      $(".text-datepicker").each ->
+        $(this).attr('type', 'text')
+
+    $(document).on "focusin click keyup change", "input[type='text'].text-datepicker", (event) ->
+      element = $(this)
+      if element.attr("autocomplete") isnt "off"
+        locale = element.attr("lang") or I18n.locale.slice(0, 2) # until we get corresponding locale codes
+        options = {}
+        $.extend options, $.datepicker.regional[locale],
+          dateFormat: "dd-mm-yy"
```

Comment:
Can we put the `dateFormat` line before the `$.datepicker.regional[locale]` one? This should ensure dates are properly localized instead of always `dd-mm-yy`.
We will end up with `dd/mm/yy`though, so maybe we need to add something like
```suggestion
  $.datepicker.regional["fr"].dateFormat = "dd-mm-yy"
```
at the top of the file ?